src/components/TopologySidebar/TopologyActionBar.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useState } from "react";
+import { IconType } from "react-icons";
+import { BiShow, BiHide, BiZoomIn } from "react-icons/bi";
+import { ImTree } from "react-icons/im";
+import { MdAlarmAdd, MdTableRows } from "react-icons/md";
+import { useNavigate } from "react-router-dom";
+import useUpdateComponentMutation from "../../api/query-hooks/mutations/useUpdateComponentMutation";
+import { EvidenceType } from "../../api/services/evidence";
+import { Topology } from "../../context/TopologyPageContext";
+import { AttachEvidenceDialog } from "../AttachEvidenceDialog";
+import { Button } from "../Button";
+import TopologySnapshotModal from "../TopologyCard/TopologySnapshotModal";
+
+type TopologyActionItem = {
+  /**
+   * The label to be shown in the action item
+   */
+  label: string;
+  /**
+   *
+   * The icon to be shown in the action item, this is a react component and can
+   * be rendered directly
+   *
+   * e.g. <Icon {...props} />
+   *
+   */
+  Icon: IconType;
```

Comment:
```suggestion
  icon: IconType;
```

Reply:
Due to how react works, components must start with a capital letter. Hence the options are either this or alias every time we need to use them in order to get intellisense and be recognized as components.